src/OrbitGl/GpuTrack.h

```diff
@@ -31,50 +33,48 @@ std::string MapGpuTimelineToTrackLabel(std::string_view timeline);
                     OrbitApp* app, const CaptureData* capture_data);
-  ~GpuTrack() override = default;
-  [[nodiscard]] std::string GetTooltip() const override;
+  void OnTimer(const orbit_client_protos::TimerInfo& timer_info);
+
+  [[nodiscard]] const TextBox* GetLeft(const TextBox* textbox) const;
+  [[nodiscard]] const TextBox* GetRight(const TextBox* textbox) const;
+
+  [[nodiscard]] const TextBox* GetUp(const TextBox* textbox) const;
+  [[nodiscard]] const TextBox* GetDown(const TextBox* textbox) const;
+
   [[nodiscard]] Type GetType() const override { return kGpuTrack; }
+  [[nodiscard]] std::string GetTooltip() const override;
   [[nodiscard]] float GetHeight() const override;
 
-  [[nodiscard]] const TextBox* GetLeft(const TextBox* text_box) const override;
-  [[nodiscard]] const TextBox* GetRight(const TextBox* text_box) const override;
-
-  [[nodiscard]] float GetYFromTimer(
-      const orbit_client_protos::TimerInfo& timer_info) const override;
+  void Draw(GlCanvas* canvas, PickingMode picking_mode, float z_offset = 0) override;
+  void UpdatePrimitives(Batcher* batcher, uint64_t min_tick, uint64_t max_tick,
+                        PickingMode picking_mode, float z_offset = 0) override;
+  [[nodiscard]] std::vector<CaptureViewElement*> GetVisibleChildren() override;
 
-  void OnTimer(const orbit_client_protos::TimerInfo& timer_info) override;
+  [[nodiscard]] std::vector<std::shared_ptr<TimerChain>> GetAllChains() const override;
+  [[nodiscard]] std::vector<std::shared_ptr<TimerChain>> GetAllSerializableChains() const override;
 
-  [[nodiscard]] bool IsCollapsible() const override {
-    return depth_ > 1 || has_vulkan_layer_command_buffer_timers_;
+  [[nodiscard]] bool IsEmpty() const override {
+    return submission_track_->IsEmpty() && marker_track_->IsEmpty();
   }
+  [[nodiscard]] bool IsCollapsible() const override { return true; }
+  void OnCollapseToggle(TriangleToggle::State state) override;
 
- protected:
-  [[nodiscard]] bool IsTimerActive(const orbit_client_protos::TimerInfo& timer) const override;
-  [[nodiscard]] Color GetTimerColor(const orbit_client_protos::TimerInfo& timer, bool is_selected,
-                                    bool is_highlighted) const override;
-  [[nodiscard]] bool TimerFilter(const orbit_client_protos::TimerInfo& timer) const override;
-  void SetTimesliceText(const orbit_client_protos::TimerInfo& timer, double elapsed_us, float min_x,
-                        float z_offset, TextBox* text_box) override;
-  [[nodiscard]] std::string GetBoxTooltip(const Batcher& batcher, PickingId id) const override;
+  [[nodiscard]] int GetVisiblePrimitiveCount() const override { return visible_timer_count_; }
 
  private:
+  void UpdatePositionOnfSubtracks();
+  const std::shared_ptr<GpuSubmissionTrack> submission_track_;
+  const std::shared_ptr<GpuDebugMarkerTrack> marker_track_;
+
+  int visible_timer_count_ = 0;
```

Comment:
Is this needed? I guess it's for the Accessibility, but it would be less error-prone to not cache this here but simply query directly from the children in `GetVisiblePrimitiveCount()`.